src/backend/utils/mmgr/mcxt_override.c

```diff
@@ -0,0 +1,172 @@
+	header->info.init = DYNAMIC_MEMORY_DEBUG_INIT_MAGIC;
+}
+
+#define MEMORY_CONTEXT_ALLOC_FUNC(__func_name__)                                                 \
+void *                                                                                           \
+_##__func_name__(MemoryContext context, Size size, const char *func, const char *file, int LINE) \
+{                                                                                                \
+	void * ret = __func_name__(context, size);                                                   \
+                                                                                                 \
+	if (ret)                                                                                     \
+		MemoryContextWriteFuncAndLineToAllocedMemory(ret, func, #__func_name__, file, LINE);     \
+                                                                                                 \
+	return ret;                                                                                  \
+}
+
+#define MEMORY_CONTEXT_PALLOC_FUNC(__func_name__)                                            \
+void *                                                                                       \
+_##__func_name__(Size size, const char *func, const char *file, int LINE)                    \
+{                                                                                            \
+	void * ret = __func_name__(size);                                                        \
+                                                                                             \
+	if (ret)                                                                                 \
+		MemoryContextWriteFuncAndLineToAllocedMemory(ret, func, #__func_name__, file, LINE); \
+                                                                                             \
+	return ret;                                                                              \
+}
+
+#define MEMORY_CONTEXT_REPALLOC_FUNC(__func_name__)                                          \
+void *                                                                                       \
+_##__func_name__(void *pointer, Size size, const char *func, const char *file, int LINE)     \
+{                                                                                            \
+	void * ret = __func_name__(pointer, size);                                               \
+                                                                                             \
+	if (ret)                                                                                 \
+		MemoryContextWriteFuncAndLineToAllocedMemory(ret, func, #__func_name__, file, LINE); \
+                                                                                             \
+	return ret;                                                                              \
+}
+
+MEMORY_CONTEXT_ALLOC_FUNC(MemoryContextAlloc)
```

Comment:
Please add a comment to describe what's going on here and why we use macros to save code lines.